### ISSUE TITLE
API for FlowGraph.find_path()

### ISSUE THREAD

**Issue by Stolas** [Vector35/binaryninja-api#3507]
**What is the feature you'd like to have?**
```
>>> type(graph)
<class 'binaryninja.flowgraph.FlowGraph'>
>>> graph.find_paths("foo", "bar")
```
I've implemented BFS, DFS, and A* for often right now for Binja that I'd just like to see it in the box.
It would be create that I could easily do something like: 
```
current_function.mlil.graph.find_paths(0x11111, 0xfffff)
```

**Is your feature request related to a problem?**
No problem, just think it would be neat and its not to hard to implement.

**Are any alternative solutions acceptable?**
I can keep using what I have now, but "batteries included" would be nice.

**Comment by psifertex** [comment]
This is a good idea but rather than implement specific algorithms we are considering adding a NetworkX compatibility layer that would let you leverage that library for the actual matching. We're guessing as to the difficulty of that compatibility shim, if it ends up being harder than "low" effort there's a good chance we decide not to implement it, but we agree it it would be a nice thing to have.

**Comment by Stolas** [comment]
I'd recommend to use a native lib with caching if you do implement this.
NetworkX (at least in my experience) is quite slow when loading larger datasets.

**Comment by psifertex** [comment]
Yeah, I've seen some poor performance there too but it is also the most popular in terms of what people are familiar with. I've read https://www.timlrx.com/blog/benchmark-of-popular-graph-network-packages-v2 but do you have any specific recommendations?

**Comment by Stolas** [comment]
After running into 'issues' (could be PEBCAK) with networkx and LEMON I decided to roll my own which can swap out bits not required at that specific moment. It works, the swap out bit I didnt write yet and at that point I thought.. Lets add this to the Binja Feature request list and make it someone else's problem ;)

**Comment by ccarpenter04** [comment]
This could be useful for pattern recognition if you're working with two different versions of the same application and would be a powerful addition in my opinion.
